source/framework/tools/src/TRestTools.cxx

```diff
@@ -211,50 +248,118 @@ template int TRestTools::ReadBinaryTable<Float_t>(string fName, std::vector<std:
 template int TRestTools::ReadBinaryTable<Double_t>(string fName, std::vector<std::vector<Double_t>>& data,
                                                    Int_t columns);
 
+///////////////////////////////////////////////
+/// \brief It identifies if the filename extension follows the formatting ".Nxyzf", where the
+/// number of columns is `xyz`, and the last character is the type of data (f/d/i), float,
+/// double and integer respectively.
+///
+Bool_t TRestTools::IsBinaryFile(string fname) {
+    if (GetBinaryFileColumns(fname) > 0) return true;
+    return false;
+}
+
+///////////////////////////////////////////////
+/// \brief It extracts the number of columns from the filename extension given by argument.
+/// The file will containing a binary formatted table with a fixed number of rows and columns.
+///
+/// The filename extension will be : ".Nxyzf", where the number of columns is `xyz`, and the
+/// last character is the type of data (f/d/i), float, double and integer respectively.
+///
+int TRestTools::GetBinaryFileColumns(string fname) {
+    string extension = GetFileNameExtension(fname);
+    if (extension.find("N") != 0) {
+        ferr << "Wrong filename extension." << endl;
+        ferr << "Cannot guess the number of columns" << endl;
+        return -1;
+    }
+
+    size_t pos = extension.find("i");
+    if (pos != string::npos) {
+        return StringToInteger(extension.substr(1, pos - 1));
+    }
+
+    pos = extension.find("f");
+    if (pos != string::npos) {
+        return StringToInteger(extension.substr(1, pos - 1));
+    }
+
+    pos = extension.find("d");
+    if (pos != string::npos) {
+        return StringToInteger(extension.substr(1, pos - 1));
+    }
+
+    ferr << "Format " << ToUpper(extension) << " not recognized" << endl;
+    return -1;
+}
+
 ///////////////////////////////////////////////
 /// \brief It returns the maximum value for a particular `column` from the table given by
-/// argument.
+/// argument. If no column is specified in the arguments, then it gets the maximum from the
+/// full table.
 ///
 /// This method is available for tables of type Float_t, Double_t and Int_t.
 ///
 template <typename T>
-T TRestTools::GetMaxValueFromTable(std::vector<std::vector<T>> data, Int_t column) {
-    if (data.size() == 0 || data[0].size() <= column) return 0;
-    T maxValue = data[0][column];
-    for (int n = 0; n < data.size(); n++)
-        if (maxValue < data[n][column]) maxValue = data[n][column];
+T TRestTools::GetMaxValueFromTable(const std::vector<std::vector<T>>& data, Int_t column) {
+    if (data.size() == 0) return 0;
+    if (column != -1 && data[0].size() <= column) return 0;
+
+    T maxValue = data[0][0];
```

Comment:
This migth be wrong, what happen if `data[0][0]` is bigger than any element on `data[0][colum]` ?

Reply:
> This migth be wrong, what happen if `data[0][0]` is bigger than any element on `data[0][column]` ?

I see, this is addressed below